src/mono/cmake/config.h.in

```diff
@@ -893,6 +893,9 @@
 /* The JIT/AOT targets tvOS */
 #cmakedefine TARGET_TVOS 1
 
+/* The JIT/AOT targets Mac Catalyst */
+#cmakedefine TARGET_MACCAT 1
```

Comment:
nit: I think we should call this `TARGET_MACCATALYST` so it's consistent with how we're using it everywhere else (can be done in a follow-up PR).

Reply:
That came in with #51669, I'm innocent